test/Dialect/FIRRTL/connect.mlir

```diff
@@ -465,3 +477,58 @@ firrtl.module @test(%a : !firrtl.uint<2>, %b : !firrtl.flip<uint<1>>) {
 }
 
 }
+
+
+/// Wire types can be on the LHS or RHS
+
+// -----
+
+firrtl.circuit "test" {
+
+firrtl.module @test(%a : !firrtl.uint<1>, %b : !firrtl.flip<uint<1>>) {
+  %w = firrtl.wire : !firrtl.uint<1>
+  // CHECK: firrtl.connect %w, %a : !firrtl.uint<1>, !firrtl.uint<1>
+  firrtl.connect %w, %a : !firrtl.uint<1>, !firrtl.uint<1>
+  // CHECK: firrtl.connect %b, %w : !firrtl.flip<uint<1>>, !firrtl.uint<1>
+  firrtl.connect %b, %w : !firrtl.flip<uint<1>>, !firrtl.uint<1>
+}
+
+}
+
+// -----
+
+firrtl.circuit "test" {
+
+firrtl.module @test(%a : !firrtl.uint<1>, %b : !firrtl.flip<uint<1>>) {
+  %w = firrtl.wire : !firrtl.flip<uint<1>>
+  // CHECK: firrtl.connect %w, %a : !firrtl.flip<uint<1>>, !firrtl.uint<1>
+  firrtl.connect %w, %a : !firrtl.flip<uint<1>>, !firrtl.uint<1>
+  // CHECK: firrtl.connect %b, %w : !firrtl.flip<uint<1>>, !firrtl.flip<uint<1>>
+  firrtl.connect %b, %w : !firrtl.flip<uint<1>>, !firrtl.flip<uint<1>>
+}
+
+}
+
+// -----
+
+firrtl.circuit "test" {
+
+firrtl.module @test(%a : !firrtl.uint<1>, %b : !firrtl.flip<uint<1>>) {
+  %w = firrtl.wire : !firrtl.uint<1>
+  // expected-error @+1 {{Connection destination must be a non-passive type or a duplex value}}
+  firrtl.connect %a, %w : !firrtl.uint<1>, !firrtl.uint<1>
+}
+
+}
+
+// -----
+
+firrtl.circuit "test" {
+
+firrtl.module @test(%a : !firrtl.uint<1>, %b : !firrtl.flip<uint<1>>) {
+  %w = firrtl.wire : !firrtl.flip<uint<1>>
+  // CHECK: firrtl.connect %w, %b  
+  firrtl.connect %w, %b : !firrtl.flip<uint<1>>, !firrtl.flip<uint<1>>
+}
```

Comment:
It looks like these `CHECK:` statements don't work when running with `-verify-diagnostics`.  I guess we are going to have to split them in to a different file.  There are a couple pre-existing ones in here that should be moved.

Reply:
Right, filecheck is what enforces CHECK: lines, so if you don't run it, they'll just be ignored.

Reply:
covered here: https://github.com/llvm/circt/pull/695